build.sbt

```diff
@@ -252,3 +230,173 @@ fixCSS := {
+  env = Map("FONTAWESOME_NPM_AUTH_TOKEN" -> "${{ secrets.FONTAWESOME_NPM_AUTH_TOKEN }}")
+)
+
+lazy val npmCache = WorkflowStep.Use(
+  UseRef.Public("c-hive", "gha-npm-cache", "v1"),
+  env = Map("FONTAWESOME_NPM_AUTH_TOKEN" -> "${{ secrets.FONTAWESOME_NPM_AUTH_TOKEN }}")
+)
+
+lazy val sbtStage = WorkflowStep.Sbt(List("stage"), name = Some("Stage"))
+
+// https://stackoverflow.com/a/55610612
+lazy val npmInstall = WorkflowStep.Run(
+  List(
+    "rm .npmrc",
+    """npm config set "@fortawesome:registry" https://npm.fontawesome.com/""",
+    """npm config set "//npm.fontawesome.com/:_authToken" ${{ secrets.FONTAWESOME_NPM_AUTH_TOKEN }}""",
+    "npm install"
+  ),
+  name = Some("npm install")
+)
+
+lazy val npmBuild = WorkflowStep.Run(
+  List("npm run build"),
+  name = Some("Build application"),
+  env = Map("NODE_OPTIONS" -> "--max-old-space-size=6144")
+)
+
+// https://frontside.com/blog/2020-05-26-github-actions-pull_request/#how-does-pull_request-affect-actionscheckout
+lazy val overrideCiCommit = WorkflowStep.Run(
+  List("""echo "CI_COMMIT_SHA=${{ github.event.pull_request.head.sha}}" >> $GITHUB_ENV"""),
+  name = Some("override CI_COMMIT_SHA"),
+  cond = Some(prCond)
+)
+
+lazy val bundlemon = WorkflowStep.Run(
+  List("yarn bundlemon"),
+  name = Some("Run BundleMon"),
+  env = Map(
+    "BUNDLEMON_PROJECT_ID"     -> "61a698e5de59ab000954f941",
+    "BUNDLEMON_PROJECT_APIKEY" -> "${{ secrets.BUNDLEMON_PROJECT_APIKEY }}"
+  )
+)
+
+def firebaseDeploy(name: String, cond: String, live: Boolean) = WorkflowStep.Use(
+  UseRef.Public("FirebaseExtended", "action-hosting-deploy", "v0"),
+  name = Some(name),
+  cond = Some(cond),
+  params = Map(
+    "repoToken"              -> "${{ secrets.GITHUB_TOKEN }}",
+    "firebaseServiceAccount" -> "${{ secrets.FIREBASE_SERVICE_ACCOUNT_EXPLORE_GEMINI }}",
+    "projectId"              -> "explore-gemini",
+    "target"                 -> "staging"
+  ) ++ (if (live) Map("channelId" -> "live") else Map.empty)
+)
+
+lazy val firebaseDeployReview = firebaseDeploy(
+  "Deploy review app to Firebase",
+  allConds(prCond,
+           notDependabotCond,
+           "github.event.pull_request.head.repo.full_name == github.repository"
+  ),
+  live = false
+)
+
+lazy val firebaseDeployStaging = firebaseDeploy(
+  "Deploy staging app to Firebase",
+  pushCond,
+  live = true
+)
+
+lazy val herokuProvision = WorkflowStep.Run(
+  List("heroku plugins:install heroku-cli-static"),
+  name = Some("Heroku - Provision static plugin")
+)
+
+lazy val herokuDeploy = WorkflowStep.Run(
+  List("cd ./heroku", "heroku static:deploy -a ${{ secrets.HEROKU_APP_NAME }}"),
+  name = Some("Heroku - Deploy"),
+  env = Map("HEROKU_API_KEY" -> "${{ secrets.HEROKU_API_KEY }}")
+)
+
+def setupVars(mode: String) = WorkflowStep.Run(
+  List(
+    raw"""sed '/^[[:blank:]]*[\\.\\}\\@]/d;/^[[:blank:]]*\..*/d;/^[[:blank:]]*$$/d;/\/\/.*/d' common/src/main/webapp/less/variables-$mode.less > common/src/main/webapp/less/vars.css""",
```

Comment:
Not sure what you mean? This one I copied from here:
https://github.com/gemini-hlsw/explore/blob/a366ec0a2be4df7a844c45ffb8da1313f258c489/.github/workflows/lint.yml#L42-L45.

Reply:
I see I forgot it was there too. It is on the package.json file too
BTW. We are having troubles with this action. We'd like it to be run for our own PRs but not for scala-steward ones but we haven't been able to get that working

Reply:
Oh, I see. Maybe we can run it as npm script instead of duplicating it here.

https://github.com/gemini-hlsw/explore/blob/a366ec0a2be4df7a844c45ffb8da1313f258c489/package.json#L64-L68

> We'd like it to be run for our own PRs but not for scala-steward ones but we haven't been able to get that working

Yeah, np I can fix that :)

Reply:
The reason is that the linter GitHub action wants to update the pr status but it fails with scala steward ones due to permissions

Reply:
Oh I see, this linting action actually commits changes and pushes. Huh.

When you make PRs, do they originate from your own repos, or do you make them from branches on explore?

Reply:
Usually branches on the main repo

Reply:
Cool, I can make that work 👍 thanks.